fragmenter/fragment.py

```diff
@@ -824,1078 +658,459 @@ class WBOFragmenter(Fragmenter):
     molecule : OEMol
         Molecule to fragment.
     functional_groups : dict, optional, default None
-        `{f_group: SMARTS}`. Dictionary that maps the name of a functional group to its SMARTS pattern.
-        These functional groups, if they exist in the molecule, will be tagged so they are not fragmented.
-        If None, will use internal list of functional group. If False, will not tag any functional groups.
+        `{f_group: SMARTS}`. Dictionary that maps the name of a functional group to its
+        SMARTS pattern. These functional groups, if they exist in the molecule, will be
+        tagged so they are not fragmented. If None, will use internal list of functional
+        group. If False, will not tag any functional groups.
 
     """
-    def __init__(self, molecule, functional_groups=None, verbose=False):
-        super().__init__(molecule)
 
-        self.verbose = verbose
+    def __init__(self, molecule, functional_groups=None, verbose=False):
 
         if functional_groups is None:
-            fn = resource_filename('fragmenter', os.path.join('data', 'fgroup_smarts.yml'))
-            with open(fn, 'r') as f:
-                functional_groups = yaml.safe_load(f)
-        self._options['functional_groups'] = functional_groups
-        self._tag_functional_groups(functional_groups)
+            functional_groups = get_fgroup_smarts()
+
+        super().__init__(molecule, functional_groups)
 
         self.rotors_wbo = {}
-        self.ring_systems = {}
-        self.fragments = {} # Fragments from fragmentation scheme for each rotatable bond
-        self._fragments = {} # Used internally for depiction
 
+        self.verbose = verbose
+        self.threshold = None
 
-    def fragment(self,  threshold=0.03, keep_non_rotor_ring_substituents=False, **kwargs):
+    def fragment(
+        self, threshold=0.03, keep_non_rotor_ring_substituents=False, **kwargs
+    ):
         """
         Fragment molecules using the Wiberg Bond Order as a surrogate
 
         Parameters
         ----------
+        threshold : float, optional, default 0.01
```

Comment:
```suggestion
        threshold : float, optional, default 0.03
```

On the whole I'm a little doubtful of documenting default values because they can change so easily, and if you can read a docstring you can usually inspect a function signature too.

Reply:
I agree wholeheartedly that default values should *not* appear in the docstring like this. My focus in this (and the child) PR was more getting the code migrated. I think after this is merged it would be good to have another pass over the docstrings as quite a few of them could probably do with a clean-up.

Reply:
Thanks for letting me know @SimonBoothroyd. There was a lot of code changes in this PR so I just looked at the new code as a whole, and wasn't sure what was in scope or not :-)